pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +521,20 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasLabel(labels map[string]string, result *newcmd.AppResult) (bool, error) {
+	for _, obj := range result.List.Items {
+		objCopy, err := kapi.Scheme.DeepCopy(obj)
+		if err != nil {
+			return false, err
+		}
+		err = util.AddObjectLabelsWithFlags(objCopy.(runtime.Object), labels, util.ErrorOnExistingDstKey)
```

Comment:
ok i thought about this some more and i guess the scenario you're thinking of is if an object already has the app label, but it's got the right value, we can go ahead and set the app label on everything else too, because there's no danger of us either:

1) replacing a label they intended to set
2) setting a mix of app labels with different values

it still feels a little debatable to me though, because you've got a template where the user made an explicit choice to put the app label on some things and not others, and now we're overriding that choice by putting the app label on everything. i think that's got potential to cause harm, so i think i still prefer the approach of "if the app label exists on any object in the template, don't add it anywhere".

wdyt?

(moved comment to correct location. whoops)